### ISSUE TITLE
Perceptor errors out

### ISSUE THREAD

**Issue by msenmurugan** [blackducksoftware/perceptor#116]
**Environment setup**
[perceptor_new.log](https://github.com/blackducksoftware/perceptor/files/1842933/perceptor_new.log)

- Since the old version of perceptor was already running on the cluster, most of the images might have already scanned by the Hub.

**Steps to reproduce**
- Uninstall the existing perceptor and install with the latest perceptor build

Number of restart = 4

Please find below logs
```
time="2018-03-23T01:30:15Z" level=info msg="INITIALIZING LOGRUS WITH METRICS HOOK [metrics subsystem = laserbeak]"
--
  | time="2018-03-23T01:30:15Z" level=info msg=start
  | time="2018-03-23T01:30:15Z" level=info msg="instantiating perceptor with config: host 35.193.103.39, user sysadmin"
  | time="2018-03-23T01:30:15Z" level=info msg="successfully got hub version 4.4.2"
  | time="2018-03-23T01:30:15Z" level=info msg="starting checking for stalled scans"
  | time="2018-03-23T01:30:15Z" level=info msg="instantiated perceptor in real mode: &{hubClient:0xc42016c120 httpResponder:0xc420064540 reducer:0xbf7838 actions:0xc420064720}"
  | time="2018-03-23T01:30:15Z" level=info msg="starting to poll hub for completed scans"
  | time="2018-03-23T01:30:35Z" level=info msg="checking hub for completed scans"
  | time="2018-03-23T01:30:36Z" level=info msg="check image in hub -- unable to find image scan for sha 725e9c13b25048b153010bdc22ac965fa2d7d4dbd16a4942e51669db678d8207, found nil"
  | time="2018-03-23T01:30:38Z" level=info msg="check image in hub -- found finished image scan for sha 6b479a1fd38b067d1c80d7db937251c977e5639e4e533ae02e88e8a68cf9234d: {Sha:6b479a1fd38b067d1c80d7db937251c977e5639e4e533ae02e88e8a68cf9234d Scan:0xc4201a6ea0 Err:<nil>}"
  | time="2018-03-23T01:30:41Z" level=info msg="check image in hub -- found finished image scan for sha 462acb1b568584fa6d84e4dd2b1ccea637f73b474ab6d0f3c87faa62aea322ab: {Sha:462acb1b568584fa6d84e4dd2b1ccea637f73b474ab6d0f3c87faa62aea322ab Scan:0xc42021eea0 Err:<nil>}"
  | time="2018-03-23T01:30:43Z" level=info msg="check image in hub -- unable to find image scan for sha 07e7cd18872eaaec985a37a837019e2a64960ebf0b5817dab0d26eadbf1544d4, found nil"
  | time="2018-03-23T01:30:45Z" level=info msg="check image in hub -- unable to find image scan for sha 2a814f56071fc4b606bcafe1c8003691daf693d806ba9e3cb15fc148edd2621f, found nil"
  | time="2018-03-23T01:30:47Z" level=info msg="check image in hub -- found finished image scan for sha e2eedf48e93cbdacdb053f223c7415472912dbde721f45ef7448e9875ddd51d5: {Sha:e2eedf48e93cbdacdb053f223c7415472912dbde721f45ef7448e9875ddd51d5 Scan:0xc4201a7930 Err:<nil>}"
  | time="2018-03-23T01:30:49Z" level=info msg="check image in hub -- unable to find image scan for sha 1d3003b89d807e812244b3aa822351bb747ecd7d96ac301f3d94ec57fa8c8ad9, found nil"
  | time="2018-03-23T01:30:51Z" level=info msg="check image in hub -- found finished image scan for sha d8209c787353407cab5e0f930f04e245ba37cb7a3a6ee7a39aff801a90a95b73: {Sha:d8209c787353407cab5e0f930f04e245ba37cb7a3a6ee7a39aff801a90a95b73 Scan:0xc42021fc70 Err:<nil>}"
  | panic: unable to remove sha d8209c787353407cab5e0f930f04e245ba37cb7a3a6ee7a39aff801a90a95b73 from hub check queue, not found
  | goroutine 12 [running]:
  | github.com/blackducksoftware/perceptor/pkg/core/model.(*Model).removeImageFromHubCheckQueue(0xc420064780, 0xc420024f40, 0x40)
  | /workspace/gopath/src/github.com/blackducksoftware/perceptor/pkg/core/model/model.go:178 +0x302
  | github.com/blackducksoftware/perceptor/pkg/core/model.(*Model).leaveState(0xc420064780, 0xc420024f40, 0x40, 0x1)
  | /workspace/gopath/src/github.com/blackducksoftware/perceptor/pkg/core/model/model.go:88 +0x8d
  | github.com/blackducksoftware/perceptor/pkg/core/model.(*Model).setImageScanStatusForSha(0xc420064780, 0xc420024f40, 0x40, 0x5, 0xc420080be0, 0xc420055dd8)
  | /workspace/gopath/src/github.com/blackducksoftware/perceptor/pkg/core/model/model.go:129 +0x109
  | github.com/blackducksoftware/perceptor/pkg/core/model.(*Model).SetImageScanStatus(0xc420064780, 0xc420024f40, 0x40, 0x5)
  | /workspace/gopath/src/github.com/blackducksoftware/perceptor/pkg/core/model/model.go:206 +0x56
  | github.com/blackducksoftware/perceptor/pkg/core/actions.(*InitialHubCheckResults).Apply(0xc42007e6b8, 0xc420064780)
  | /workspace/gopath/src/github.com/blackducksoftware/perceptor/pkg/core/actions/initialhubcheckresults.go:68 +0x390
  | github.com/blackducksoftware/perceptor/pkg/core.newReducer.func1(0xc420064720, 0xc42000ba60, 0xc420064780)
  | /workspace/gopath/src/github.com/blackducksoftware/perceptor/pkg/core/reducer.go:54 +0x1ff
  | created by github.com/blackducksoftware/perceptor/pkg/core.newReducer
  | /workspace/gopath/src/github.com/blackducksoftware/perceptor/pkg/core/reducer.go:39 +0x9d
```

**Comment by mattfenwick** [comment]
I think this is fixed by #117 , I'll be doing some more testing to verify.

**Comment by mattfenwick** [comment]
Looks like this is fixed!
